File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.js
const addType = (s1, s2, personalityType) => {
  if (personalityType[s1] > personalityType[s2]) {
    return s1;
  } else if (personalityType[s1] < personalityType[s2]) {
    return s2;
  } else {
    return [s1, s2].sort()[0];
  }
};

function solution(survey, choices) {
  const personalityType = {
    R: 0,
    T: 0,
    C: 0,
    F: 0,
    J: 0,
    M: 0,
    A: 0,
    N: 0,
  };

  const point = [0, 3, 2, 1, 0, 1, 2, 3];

  for (let i = 0; i < survey.length; i++) {
    // 오른쪽 유형에 점수 주기
    if (choices[i] > 4) {
      personalityType[survey[i][1]] += point[choices[i]];
      // 왼쪽 유형에 점수 주기
    } else if (choices[i] < 4) {
      personalityType[survey[i][0]] += point[choices[i]];
    }
  }

  let ans = "";

  ans += addType("R", "T", personalityType);
  ans += addType("C", "F", personalityType);
  ans += addType("J", "M", personalityType);
  ans += addType("A", "N", personalityType);

  return ans;
}

File: 프로그래머스/lv1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    answer = ''
    
    obj = {
        "R": 0, "T": 0, "C": 0, "F": 0, "J": 0, "M": 0, "A": 0, "N": 0
    }
    
    for i in range(len(survey)):
        if choices[i] == 1:
            obj[survey[i][0]] += 3
        elif choices[i] == 2:
            obj[survey[i][0]] += 2
        elif choices[i] == 3:
            obj[survey[i][0]] += 1
        elif choices[i] == 5:
            obj[survey[i][1]] += 1
        elif choices[i] == 6:
            obj[survey[i][1]] += 2
        elif choices[i] == 7:
            obj[survey[i][1]] += 3
            
    ref = ["R", "T", "C", "F", "J", "M", "A", "N"]
    for i in range(0, 8, 2):
        if obj[ref[i]] > obj[ref[i + 1]]:
            answer += ref[i]
        elif obj[ref[i]] < obj[ref[i + 1]]:
            answer += ref[i + 1]
        else:
            tmp = ref[i] + ref[i + 1]
            answer += tmp[0]
        
    return answer